public/app/features/alerting/unified/components/GrafanaAlertmanagerDeliveryWarning.tsx

```diff
@@ -5,37 +5,58 @@ import { GrafanaTheme2 } from '@grafana/data/src';
 import { Alert, useStyles2 } from '@grafana/ui/src';
 
 import { AlertmanagerChoice } from '../../../../plugins/datasource/alertmanager/types';
+import { alertmanagerApi } from '../api/alertmanagerApi';
 import { GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
 
 interface GrafanaAlertmanagerDeliveryWarningProps {
-  alertmanagerChoice?: AlertmanagerChoice;
   currentAlertmanager: string;
 }
 
-export function GrafanaAlertmanagerDeliveryWarning({
-  alertmanagerChoice,
-  currentAlertmanager,
-}: GrafanaAlertmanagerDeliveryWarningProps) {
+export function GrafanaAlertmanagerDeliveryWarning({ currentAlertmanager }: GrafanaAlertmanagerDeliveryWarningProps) {
   const styles = useStyles2(getStyles);
 
-  if (currentAlertmanager !== GRAFANA_RULES_SOURCE_NAME) {
+  const { useGetAlertmanagerChoiceStatusQuery } = alertmanagerApi;
+  const { currentData: amChoiceStatus } = useGetAlertmanagerChoiceStatusQuery();
+
+  const viewingInternalAM = currentAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+
+  const interactsWithExternalAMs =
+    amChoiceStatus?.alertmanagersChoice &&
+    [AlertmanagerChoice.External, AlertmanagerChoice.All].includes(amChoiceStatus?.alertmanagersChoice);
+
+  if (!interactsWithExternalAMs || !viewingInternalAM) {
     return null;
   }
 
-  if (alertmanagerChoice !== AlertmanagerChoice.External) {
-    return null;
+  const hasActiveExternalAMs = amChoiceStatus.numExternalAlertmanagers > 0;
+
+  if (amChoiceStatus.alertmanagersChoice === AlertmanagerChoice.External && hasActiveExternalAMs) {
```

Comment:
If `alertmanagersChoice === External` and `hasActiveExternalAMs === 0` there is still no point in configuring Grafana Alertmanager, isn't it?

Reply:
Good point, I don't think we need to check how many external AMs are configured 🤔